tests/sentry/api/endpoints/test_sentry_app_details.py

```diff
@@ -147,6 +147,20 @@ def test_update_unpublished_app(self):
         assert response.data['uuid'] == self.unpublished_app.uuid
         assert response.data['webhookUrl'] == 'https://newurl.com'
 
+    @with_feature('organizations:sentry-apps')
+    def test_cannot_update_name_with_non_unique_slug(self):
+        self.login_as(user=self.user)
+        response = self.client.put(
+            self.url,
+            data={
+                'name': 'test',
+            },
+            format='json',
+        )
+        assert response.status_code == 400
+        assert response.data == \
+            {"name": ["Name test is already taken, please use another."]}
```

Comment:
Let's add a test where it compares the name/slug against a deleted app. Currently, I don't think it'd pass based on the comment above.